### PR TITLE
Math Magicians: refactor with hooks

### PR DESCRIPTION
**Pull request by Mhdez221993**
# Project requirements

* Refactor the Calculator component (components/Calculator.js) from a class based component to a functional component with hooks.
* After the refactor is done, test the app in the browser. It should keep the same functionality.

### CODE REVIEW

File: src/components/Screen.js [hunk 1]
```diff
@@ -2,22 +2,16 @@
 /* eslint-disable react/prop-types */
 import React from 'react';
```
Comment (henatan99) [inline src/components/Screen.js]: - [x] Kindly do not disable `eslint` rules as they are there to improve your code quality. :pray:

File: src/components/Calculator.js [hunk 2]
```diff
@@ -1,40 +1,31 @@
...
-}
+  const {
+    next, total, operation, equalSign,
+  } = state;
+  const curr = next || total || '0';
+  const currOperation = getCurrOperation([total, operation, next, equalSign]);
+  return (
+    <div className="Calculator">
+      <Screen value={[currOperation]} />
+      <Screen value={curr} />
```
Comment (henatan99) [inline src/components/Calculator.js]: - [x] I don't know if you need to render the `Screen` component twice and it is a bit vague what you wanted to achieve with it. But, it compromised the UI as it gives a different image than the original design. Kindly fix this issue! :pray:

image

**Review by henatan99 — CHANGES_REQUESTED** [review]
Hi @Mhdez221993 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- [ ] OPTIONAL: Please try to commit your changes as often as possible. It helps track your changes easily. :pray: 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Calculator.js (hunk 2 above)
Comment (Mhdez221993) [reply src/components/Calculator.js:25]: Thank you for the comment, I just wanted to make it look like a real calculator.

**Review by barackm — APPROVED** [review]
Hi @Mhdez221993, 👋 

# STATUS: Approved  🎉 
Your project is complete!  There is nothing else to say other than... it's time to merge it
Congratulations! 🎉  🎉 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**